### PR TITLE
Refactor logging

### PR DESCRIPTION
**Pull request by mrT23**
## Goals
- divide the log levels into a simple info/debug levels
- log _every relevant detail_ from a run, to ensure each run can be properly debugged, in an easy and accessible way:

  - relevant configuration
  - input patches
  - AI prompt
  - AI text response
  - AI response extra details
  - prev and new labels
  - actual answer sent back to github
  - more
- avoid generic log messages without actual data, to reduce clutter

___

## **Type**
enhancement, documentation

___

## **Description**
- Enhanced logging across various modules for better readability and debugging.
- Introduced dynamic log level setting based on environment variable.
- Adjusted logging levels and added debug logs for relevant configurations and outputs.
- Improved handling and logging for user descriptions in Git provider.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>13 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>litellm_ai_handler.py</strong><dd><code>Enhance AI Handler Logging</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/algo/ai_handlers/litellm_ai_handler.py
<li>Added debug logging for prompts and full AI response.<br> <li> Removed logging of <code>usage</code> from AI response.

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/708/files#diff-ea1acaa0907f3410665530fbc4cda2ab524de2772e0bbe10bad4648b8be35dfe">+5/-3</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pr_processing.py</strong><dd><code>Improve PR Processing Logging</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/algo/pr_processing.py
<li>Added info logging for filtered files, PR main language, tokens count, <br>and pruning details.<br> <li> Enhanced debug logging for the final diff after pruning.

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/708/files#diff-c2d64de584ef3cb05c9007d72137f6c7ce04c9bc23ce1931760af3a568edb04e">+29/-7</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>utils.py</strong><dd><code>Adjust Logging Level for User Labels</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/algo/utils.py
- Changed logging level for keeping user labels from info to debug.

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/708/files#diff-6b9df72d53c6f0d89fb142c210238a276c0782305e0024d16fbfcaf72c2e2b53">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>git_provider.py</strong><dd><code>Enhance Git Provider Logging and Handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/git_providers/git_provider.py
<li>Added handling for user_description attribute.<br> <li> Changed logging level for existing description to debug.

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/708/files#diff-52d45f12b836f77ed1aef86e972e65404634ea4e2a6083fb71a9b0f9bb9e062f">+7/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>utils.py</strong><dd><code>Improve Logging for Repo Settings Application</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/git_providers/utils.py
- Improved logging for applying repo settings.

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/708/files#diff-314199df3610cf35a472e952294faf0f5fbb516a2bd217e891a510fd1a698bca">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>__init__.py</strong><dd><code>Refine Logging Setup</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/log/__init__.py
- Adjusted logging setup for JSON format and console output.

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/708/files#diff-a560f2add42b09865c75e7e57a174c49069a560bcefeb6f8c89647f4316f4294">+3/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>github_app.py</strong><dd><code>Enhance Logging in GitHub App Server</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/servers/github_app.py
- Enhanced logging for request body in various event handling.

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/708/files#diff-3fb3f174152732d1b69953c8bd81b8a0a30f0e42100dc626d932da8371851608">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pr_code_suggestions.py</strong><dd><code>Improve Logging in PR Code Suggestions Tool</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/tools/pr_code_suggestions.py
<li>Added debug logging for relevant configs and PR output.<br> <li> Removed verbose info logging.

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/708/files#diff-b57ba775e741d6f80bc4f8154b71330c011dae0ac43f3d0197e785b3e6b7117b">+39/-47</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pr_description.py</strong><dd><code>Improve Logging in PR Description Tool</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/tools/pr_description.py
<li>Added debug logging for relevant configs and PR output.<br> <li> Adjusted logging for error handling and configurations.

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/708/files#diff-66130451adce278a098a5770e2db52b12050205fcd4d80c13f813615d4449abb">+22/-40</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pr_help_message.py</strong><dd><code>Improve Logging in PR Help Message Tool</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/tools/pr_help_message.py
- Added debug logging for relevant configs and PR output.

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/708/files#diff-6f30977025b20a27640fc33cfe2db79160753bd30c7879f60f2b5894d86f78de">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pr_questions.py</strong><dd><code>Improve Logging in PR Questions Tool</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/tools/pr_questions.py
- Added debug logging for relevant configs and PR output.

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/708/files#diff-71970fafa37a18dfb27eb7b47754bfa7841e14686b181445298f914e03fc31aa">+13/-10</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pr_reviewer.py</strong><dd><code>Improve Logging in PR Reviewer Tool</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/tools/pr_reviewer.py
<li>Added debug logging for relevant configs and PR output.<br> <li> Enhanced AI response logging.

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/708/files#diff-8e265068e189a06852605cc694b03e92b523b4f8162077a2f3455ced4cdad8dc">+12/-26</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pr_update_changelog.py</strong><dd><code>Improve Logging in PR Update Changelog Tool</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/tools/pr_update_changelog.py
- Added debug logging for relevant configs and PR output.

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/708/files#diff-76d8133da16132fa69d308bf1e5edb4fc7de18343a4d99760960416c54c79087">+14/-15</a>&nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>cli.py</strong><dd><code>Dynamic Log Level Configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/cli.py
- Introduced dynamic log level setting based on environment variable.

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/708/files#diff-88577a9bac5c261b52dcec31fdf5e525cef5c6ddffdcb6bde0e9ba183fa05996">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by mrT23** [comment]
Preparing PR description...

**Comment by mrT23** [comment]
Preparing PR description...

**Comment by mrT23** [comment]
Preparing PR description...

**Comment by mrT23** [comment]
Preparing PR description...

**Comment by mrT23** [comment]
Preparing PR description...

**Comment by mrT23** [comment]
Preparing PR description...

**Comment by mrT23** [comment]
**[PR Description](https://github.com/Codium-ai/pr-agent/pull/708)** updated to latest commit (https://github.com/Codium-ai/pr-agent/commit/7dcbdffe3ade9dd9a95956ceb036fb52373b856d)

**Comment by mrT23** [comment]
Preparing PR description...

**Comment by mrT23** [comment]
**[PR Description](https://github.com/Codium-ai/pr-agent/pull/708)** updated to latest commit (https://github.com/Codium-ai/pr-agent/commit/7dcbdffe3ade9dd9a95956ceb036fb52373b856d)

**Comment by mrT23** [comment]
Preparing PR description...

**Comment by mrT23** [comment]
**[PR Description](https://github.com/Codium-ai/pr-agent/pull/708)** updated to latest commit (https://github.com/Codium-ai/pr-agent/commit/ba96e147c7874aab0d47299c8877983491bd8be8)

**Comment by mrT23** [comment]
Preparing PR description...

**Comment by mrT23** [comment]
**[PR Description](https://github.com/Codium-ai/pr-agent/pull/708)** updated to latest commit (https://github.com/Codium-ai/pr-agent/commit/877796b539c1616521e461cfc1911154ddecd9b1)

**Comment by mrT23** [comment]
## Generating PR code suggestions


Work in progress ...<br>

**Comment by mrT23** [comment]
## Generating PR code suggestions


Work in progress ...<br>

**Comment by mrT23** [comment]
## Generating PR code suggestions


Work in progress ...<br>

**Comment by mrT23** [comment]
## Generating PR code suggestions


Work in progress ...<br>

File: pr_agent/algo/ai_handlers/litellm_ai_handler.py [hunk 1]
```diff
@@ -113,6 +114,8 @@ async def chat_completion(self, model: str, system: str, user: str, temperature:
             }
             if self.aws_bedrock_client:
                 kwargs["aws_bedrock_client"] = self.aws_bedrock_client
+
+            get_logger().debug(f"Prompts", system=system, user=user)
```
Comment (mrT23) [inline pr_agent/algo/ai_handlers/litellm_ai_handler.py:118]: /ask is this a valid log line ? (loguru)

File: pr_agent/algo/ai_handlers/litellm_ai_handler.py [hunk 2]
```diff
@@ -113,6 +114,8 @@ async def chat_completion(self, model: str, system: str, user: str, temperature:
             }
             if self.aws_bedrock_client:
                 kwargs["aws_bedrock_client"] = self.aws_bedrock_client
+
+            get_logger().debug(f"Prompts", messages={"system": system, "user": user})
```
Comment (mrT23) [inline pr_agent/algo/ai_handlers/litellm_ai_handler.py:118]: /ask is this a valid log line ?

**Review by okotek — APPROVED** [review]

**Pull request merged**